## 1. Fonte dos dados

Wind & Solar Energy Production Dataset

- Conjunto de dados: https://www.kaggle.com/datasets/ahmeduzaki/wind-and-solar-energy-production-dataset?resource=download

Produção de energia eólica e solar, permitindo comparar o comportamento das duas fontes renováveis.

## 2. Bibliotecas

In [7]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Situação

Um operador de um portfólio de fontes renováveis deseja comparar a ocorrência de períodos de alta produção solar e alta produção eólica. Como as duas fontes possuem escalas próprias, cada uma deverá ser comparada com o seu próprio valor máximo.

# Etapa A — Orange Data Mining

1. Carregue o dataset e visualize os atributos disponíveis.

2. Utilize Select Columns para manter os atributos temporais e as variáveis de produção solar e eólica.

3. Observe os valores mínimos e máximos das duas variáveis de geração.

4. Verifique a existência de valores ausentes.

5. Gere uma amostra aleatória de 20% dos registros.

6. Exporte a amostra em CSV.

In [18]:
# Amostra gerada no Orange
dados = pd.read_csv("/content/amostra_producao_energia.csv")

print("--- Primeiros Registros ---")
print(dados.head())

--- Primeiros Registros ---
         Date  Start_Hour  End_Hour Source  Production
0  12/18/2022          21        22   Wind       11400
1   1/29/2024          14        15   Wind        7917
2    8/1/2024          15        16  Solar        8835
3  11/10/2020           1         2   Wind         989
4  12/19/2022           1         2   Wind       12526


# Etapa B — Python / Pandas

1. Renomeie as variáveis principais para Geracao_Solar e Geracao_Eolica.

2. Determine o valor máximo registrado para cada fonte.

3. Calcule separadamente 70% do máximo solar e 70% do máximo eólico.

4. Crie um DataFrame para registros de alta geração solar e outro para registros de alta geração eólica.

5. Conte os registros de cada DataFrame e calcule seus respectivos percentuais.

6. Compare qual fonte aparece com maior frequência acima de 70% do seu próprio máximo.

7. Explique por que não seria adequado utilizar diretamente o mesmo valor numérico de potência como limite para as duas fontes sem antes observar a escala de cada variável.

In [25]:
# 1. Renomeie as variáveis principais para Geracao_Solar e Geracao_Eolica.
dados_renomeados = dados.copy()

# Criar colunas para Geracao_Solar e Geracao_Eolica baseadas na coluna 'Source'
dados_renomeados['Geracao_Solar'] = dados_renomeados.apply(
    lambda row: row['Production'] if row['Source'] == 'Solar' else 0, axis=1
)
dados_renomeados['Geracao_Eolica'] = dados_renomeados.apply(
    lambda row: row['Production'] if row['Source'] == 'Wind' else 0, axis=1
)

print("--- Primeiros Registros com Novas Colunas de Geração ---")
display(dados_renomeados.head())

--- Primeiros Registros com Novas Colunas de Geração ---


,Date,Start_Hour,End_Hour,Source,Production,Geracao_Solar,Geracao_Eolica
0,12/18/2022,21,22,Wind,11400,0,11400
1,1/29/2024,14,15,Wind,7917,0,7917
2,8/1/2024,15,16,Solar,8835,8835,0
3,11/10/2020,1,2,Wind,989,0,989
4,12/19/2022,1,2,Wind,12526,0,12526


In [20]:
# 2. Determine o valor máximo registrado para cada fonte.
max_geracao_solar = dados_renomeados['Geracao_Solar'].max()
max_geracao_eolica = dados_renomeados['Geracao_Eolica'].max()

print(f"\nValor Máximo de Geração Solar: {max_geracao_solar}")
print(f"Valor Máximo de Geração Eólica: {max_geracao_eolica}")


Valor Máximo de Geração Solar: 16578
Valor Máximo de Geração Eólica: 23446


In [21]:
# 3. Calcule separadamente 70% do máximo solar e 70% do máximo eólico.
setenta_porcento_solar = max_geracao_solar * 0.70
setenta_porcento_eolica = max_geracao_eolica * 0.70

print(f"\n70% do Máximo Solar: {setenta_porcento_solar:.2f}")
print(f"70% do Máximo Eólico: {setenta_porcento_eolica:.2f}")


70% do Máximo Solar: 11604.60
70% do Máximo Eólico: 16412.20


In [22]:
# 4. Crie um DataFrame para registros de alta geração solar e outro para registros de alta geração eólica.
alta_geracao_solar_df = dados_renomeados[dados_renomeados['Geracao_Solar'] > setenta_porcento_solar].copy()
alta_geracao_eolica_df = dados_renomeados[dados_renomeados['Geracao_Eolica'] > setenta_porcento_eolica].copy()

print("\n--- Registros de Alta Geração Solar (Primeiros 5) ---")
display(alta_geracao_solar_df.head())
print(f"Total de registros de alta geração solar: {alta_geracao_solar_df.shape[0]}")

print("\n--- Registros de Alta Geração Eólica (Primeiros 5) ---")
display(alta_geracao_eolica_df.head())
print(f"Total de registros de alta geração eólica: {alta_geracao_eolica_df.shape[0]}")


--- Registros de Alta Geração Solar (Primeiros 5) ---


,Date,Start_Hour,End_Hour,Source,Production,Geracao_Solar,Geracao_Eolica
1035,10/18/2025,12,13,Solar,12358,12358,0
1036,10/18/2025,13,14,Solar,12634,12634,0
1153,10/13/2025,15,16,Solar,12023,12023,0
1161,10/13/2025,13,14,Solar,12597,12597,0
1162,10/13/2025,14,15,Solar,12354,12354,0


Total de registros de alta geração solar: 167

--- Registros de Alta Geração Eólica (Primeiros 5) ---


,Date,Start_Hour,End_Hour,Source,Production,Geracao_Solar,Geracao_Eolica
678,11/2/2025,13,14,Wind,17903,0,17903
681,11/2/2025,12,13,Wind,17124,0,17124
691,11/2/2025,14,15,Wind,17067,0,17067
818,10/27/2025,15,16,Wind,18928,0,18928
819,10/27/2025,13,14,Wind,22044,0,22044


Total de registros de alta geração eólica: 1018


In [23]:
# 5. Conte os registros de cada DataFrame e calcule seus respectivos percentuais.

total_registros = dados_renomeados.shape[0]

count_alta_solar = alta_geracao_solar_df.shape[0]
percent_alta_solar = (count_alta_solar / total_registros) * 100

count_alta_eolica = alta_geracao_eolica_df.shape[0]
percent_alta_eolica = (count_alta_eolica / total_registros) * 100

print(f"\nNúmero de registros com alta geração solar: {count_alta_solar} ({percent_alta_solar:.2f}% do total)")
print(f"Número de registros com alta geração eólica: {count_alta_eolica} ({percent_alta_eolica:.2f}% do total)")


Número de registros com alta geração solar: 167 (0.32% do total)
Número de registros com alta geração eólica: 1018 (1.96% do total)


In [24]:
# 6. Compare qual fonte aparece com maior frequência acima de 70% do seu próprio máximo.
if count_alta_solar > count_alta_eolica:
    print("\nA geração solar ocorre com maior frequência acima de 70% do seu próprio máximo.")
elif count_alta_eolica > count_alta_solar:
    print("\nA geração eólica ocorre com maior frequência acima de 70% do seu próprio máximo.")
else:
    print("\nA geração solar e eólica ocorrem com frequência semelhante acima de 70% do seu próprio máximo.")


A geração eólica ocorre com maior frequência acima de 70% do seu próprio máximo.


## 7. Explique por que não seria adequado utilizar diretamente o mesmo valor numérico de potência como limite para as duas fontes sem antes observar a escala de cada variável.

Usar o mesmo valor numérico de potência como limite fixo para solar e eólica seria inadequado porque as duas fontes têm escalas de produção diferentes (por exemplo, o máximo eólico de 23446 supera o máximo solar de 16578) e perfis de geração intrínsecos distintos, ligados a fatores ambientais diferentes (radiação solar vs. velocidade do vento); assim, um limite absoluto representaria porcentagens muito diferentes do potencial máximo de cada fonte, distorcendo a comparação, enquanto uma métrica relativa — como 70% do próprio máximo de cada fonte — permite avaliar o quão perto cada uma está de sua capacidade individual, tornando a análise de "alta produção" justa e significativa, já que esse conceito depende do potencial específico de cada tecnologia e não de um ponto arbitrário numa escala comum.